src/Build/BackEnd/BuildManager/BuildRequestData.cs

```diff
@@ -221,6 +222,170 @@ public BuildRequestData(string projectFullPath, IDictionary<string, string> glob
+        {
+        }
+
+        /// <summary>
+        /// Constructs a BuildRequestData for build requests based on a project graph entry points.
+        /// </summary>
+        /// <param name="projectGraphEntryPoints">The entry points to use in the build.</param>
+        /// <param name="targetsToBuild">The targets to build.</param>
+        public BuildRequestData(IEnumerable<ProjectGraphEntryPoint> projectGraphEntryPoints, string[] targetsToBuild)
+            : this(projectGraphEntryPoints, targetsToBuild, null, BuildRequestDataFlags.None, null)
+        {
+        }
+
+        /// <summary>
+        /// Constructs a BuildRequestData for build requests based on a project graph entry points.
+        /// </summary>
+        /// <param name="projectGraphEntryPoints">The entry points to use in the build.</param>
+        /// <param name="targetsToBuild">The targets to build.</param>
+        /// <param name="hostServices">The host services to use, if any.  May be null.</param>
+        public BuildRequestData(IEnumerable<ProjectGraphEntryPoint> projectGraphEntryPoints, string[] targetsToBuild, HostServices hostServices)
+            : this(projectGraphEntryPoints, targetsToBuild, hostServices, BuildRequestDataFlags.None, null)
+        {
+        }
+
+        /// <summary>
+        /// Constructs a BuildRequestData for build requests based on a project graph entry points.
+        /// </summary>
+        /// <param name="projectGraphEntryPoints">The entry points to use in the build.</param>
+        /// <param name="targetsToBuild">The targets to build.</param>
+        /// <param name="hostServices">The host services to use, if any.  May be null.</param>
+        /// <param name="flags">Flags controlling this build request.</param>
+        public BuildRequestData(IEnumerable<ProjectGraphEntryPoint> projectGraphEntryPoints, string[] targetsToBuild, HostServices hostServices, BuildRequestDataFlags flags)
+            : this(projectGraphEntryPoints, targetsToBuild, hostServices, flags, null)
+        {
+        }
+
+        /// <summary>
+        /// Constructs a BuildRequestData for build requests based on a project graph entry points.
+        /// </summary>
+        /// <param name="projectGraphEntryPoints">The entry points to use in the build.</param>
+        /// <param name="targetsToBuild">The targets to build.</param>
+        /// <param name="hostServices">The host services to use, if any.  May be null.</param>
+        /// <param name="flags">Flags controlling this build request.</param>
+        /// <param name="propertiesToTransfer">The list of properties whose values should be transferred from the project to any out-of-proc node.</param>
+        public BuildRequestData(IEnumerable<ProjectGraphEntryPoint> projectGraphEntryPoints, string[] targetsToBuild, HostServices hostServices, BuildRequestDataFlags flags, IEnumerable<string> propertiesToTransfer)
```

Comment:
Some of the build request data flags and that `propertiesToTransfer` sound like they either don't apply that well to graph builds or they would need extra consideration. If that's the case, there three options:
- make sure all the options have some determined behaviour for graph builds
- not care about these for now and not provide the overloads